identity/src/androidTest/java/com/android/identity/IdentityCredentialTest.java

```diff
@@ -0,0 +1,172 @@
+    checkStaticAuthData(100, 100); // 100 keys x 100 Bytes = 10 KBytes
+    checkStaticAuthData(1, 100000); // 1 key x 100 KBytes
+  }
+
+  @LargeTest
+  @Test
+  public void storeStaticAuthenticationDataLarge()
+          throws IdentityCredentialException {
+    Assume.assumeTrue("Test fails on API <= 28", VERSION.SDK_INT >= 29 /* Android 10+ */);
+    // Per b/234563696, I've confirmed these failing on:
+    // Pixel 2 API 24
+    // Pixel 2 API 26
+    // Pixel 3 API 28
+    checkStaticAuthData(10, 10000); // fails
+    checkStaticAuthData(100, 1000); // flaky
+    checkStaticAuthData(100, 10000); // fails
+  }
+
+  private void checkStaticAuthData(int numAuthKeys, int staticAuthDataSizeBytes)
+          throws IdentityCredentialException {
+    final int usesPerKey = 3;
+    ProvisioningTest.createCredential(store, CREDENTIAL_NAME);
+    try {
+      IdentityCredential credential = Preconditions.checkNotNull(getCredentialIfPresent());
+
+      storeFakeStaticAuthData(credential, usesPerKey, numAuthKeys, staticAuthDataSizeBytes);
+
+      // Check that this doesn't throw (it did throw in internal b/234563696).
+      try {
+        getCredentialIfPresent();
+      } catch (RuntimeException e) {
+        String msg = String.format(Locale.US,
+                "Failed to load credential for %d authKeys with %d bytes static auth data each",
+                numAuthKeys, staticAuthDataSizeBytes);
+        throw new IdentityCredentialException(msg, e);
+      }
+    } finally {
+      deleteCredentialFromStoreIfPresent();
+    }
+  }
+
+  @Before
+  public void setUp() {
+    deleteCredentialFromStoreIfPresent(); // clean up in case a previous test didn't
+  }
+
+  @After
+  public void tearDown() {
+    deleteCredentialFromStoreIfPresent();
+    store = null;
+  }
+
+  private static void storeFakeStaticAuthData(IdentityCredential credential,
+          int maxUsesPerKey, int numKeys, int staticAuthDataSizeBytes) {
+    Random random = new Random(31337 + numKeys);
+    credential.setAvailableAuthenticationKeys(numKeys, maxUsesPerKey);
+    Collection<X509Certificate> authKeys = credential.getAuthKeysNeedingCertification();
+    assertEquals(numKeys, authKeys.size());
+
+    for (X509Certificate authKey : authKeys) {
+      byte[] staticAuthData = new byte[staticAuthDataSizeBytes];
+      random.nextBytes(staticAuthData);
+      Calendar expirationDate = Calendar.getInstance();
+      expirationDate.add(Calendar.YEAR, 1);
+      try {
+        credential.storeStaticAuthenticationData(authKey, expirationDate, staticAuthData);
+      } catch (UnknownAuthenticationKeyException e) {
+        throw new AssertionFailedError("Auth authKey not recognized: " + e);
+      }
+    }
+  }
+
+  private @Nullable IdentityCredential getCredentialIfPresent() {
+    try {
+      return store.getCredentialByName(CREDENTIAL_NAME, CIPHER_SUITE);
+    } catch (CipherSuiteNotSupportedException e) {
+      throw new AssertionFailedError("Unexpected exception: " + e);
```

Comment:
Why wrap the exception?

IIUC, if we let the original exception bubble up, the test framework will mark the test as failed and print the call stack for us.

IMO it would be a lot more readable to simply call `getCredentialByName` in the call sites instead of `getCredentialIfPresent`, as the wrapper is simply an alias at this point.

Reply:
Done.

Personally I like extracting such small helper methods because they de-duplicate the choice of CREDENTIAL_NAME, CIPHER_SUITE better as a logic unit than merely extracting the constants does.

The throw clause was left over from when this code was inside a larger method and the compiler didn't know that code after it wouldn't run. AssertionFailedError is what Assert.fail() will always throw, so "new AssertionFailedError" is the idiom of expressing the same as fail() but in a way that proves to the compiler that the line will always throw.